src/aws/aws.service.ts

```diff
@@ -0,0 +1,49 @@
+import { BadRequestException, Injectable } from '@nestjs/common';
+import { UploadURLDto } from './dto/upload-URL.dto';
+import { S3 } from 'aws-sdk';
+import { ApiConfigService } from '../../common/config/api-config.service';
+
+@Injectable()
+export class AwsService {
+    bucketName = this.configService.AWS.bucketName;
+    constructor(private readonly configService: ApiConfigService) {}
+
+    async getPreSignedUrl(
+        action: string,
+        objectKey: string,
+        contentType?: object,
+    ): Promise<string> {
+        try {
+            const params = {
+                Bucket: this.bucketName,
+                Key: objectKey,
+                Expires: this.configService.AWS.expires,
+                ...contentType,
+            };
+            const s3 = new S3({
+                accessKeyId: this.configService.AWS.accessKeyId,
+                secretAccessKey: this.configService.AWS.secretAccessKey,
+                region: this.configService.AWS.region,
+                signatureVersion: this.configService.AWS.signatureVersion,
+            });
+            return s3.getSignedUrlPromise(action, params);
```

Comment:
`s3.getSignedUrlPromise` function is async and will not throw an exception without `await`.
Please update this line as follows:

`return await s3.getSignedUrlPromise(action, params);`